packages/base/source/adt/union/union.js

```diff
@@ -82,8 +108,11 @@ instead to check if a value belongs to the ADT variant.`);
        *   where 'b = 'a[`@@folktale:adt:tag]
        */
       matchWith(pattern) {
-        return pattern[name](this);
-      } 
+        assertObject('Union.mapObject#matchWith', pattern);
+        if (name in pattern) { return pattern[name](this); }
+        else if (ANY in pattern) { return pattern[ANY](this) }
```

Comment:
Now this test is failing:

```javascript
it('any works with all nested variants', ()=> {
        const { Miss, Hit, Attack } = union('', { 
          Miss() { return { } },
          Hit(damage, critical) { return { damage, critical } },
          Attack(attackResult) { return { attackResult } }
        });

         $ASSERT(
           Attack(Hit(1, false)).matchWith({
            Miss: ()=> 'miss',
            Hit: ({damage, critical}) => 'damage',
            [any]: ({ attackResult }) => attackResult.damage
          })
          == 1
        );
      });
```

Trying to figure out why; I rewrote the [any] to take just a plain argument, but it's undefined. Is this because the payload was lost without passing in this?